### PR TITLE
[TRA-86] scaffold x/vault

### PR DESCRIPTION
**Pull request by tqin7**
### Changelist
scaffold `x/vault` 

### Test Plan
[Describe how this PR was tested (if applicable)]

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

### CODE REVIEW

File: protocol/app/testdata/default_genesis_state.json [hunk 1]
```diff
@@ -454,6 +454,7 @@
     "total_escrowed": []
   },
   "upgrade": {},
+  "vault": {},
```
Comment (Christopher-Li) [inline protocol/app/testdata/default_genesis_state.json:457]: do we even need to add this to genesis? It seems like at genesis on mainnet/testnet, we didn't have a vault, so wouldn't we default to `{}` for vault?

**Review by Christopher-Li — COMMENTED** [review]
IIUC this module has 0 business logic right? It's just an empty module that you will fill as time goes on?

**Comment by tqin7** [comment]
> IIUC this module has 0 business logic right? It's just an empty module that you will fill as time goes on?

yep just an empty module for now and will add business logic later on

File: protocol/app/testdata/default_genesis_state.json (hunk 1 above)
Comment (tqin7) [reply protocol/app/testdata/default_genesis_state.json:457]: this is needed for tests on default genesis state. added a new module so genesis state has this new entry

**Review by Christopher-Li — APPROVED** [review]

**Review by vincentwschau — APPROVED** [review]

File: protocol/testing/containertest/preupgrade_genesis.json [hunk 2]
```diff
@@ -2243,6 +2243,7 @@
         "total_escrowed": []
       },
       "upgrade": {},
+      "vault": {},
```
Comment (roy-dydx) [inline protocol/testing/containertest/preupgrade_genesis.json:2246]: Remove this

**Pull request merged**